pages/posts/[filename].tsx

```diff
@@ -76,7 +76,7 @@ export const getStaticPaths = async () => {
     paths: postsListData.getPostsList.edges.map((post) => ({
       params: { filename: post.node.sys.filename },
     })),
-    fallback: false,
+    fallback: true,
```

Comment:
```suggestion
    fallback:'blocking',
```
`blocking` will block server-side instead of client-side, and will prevent the flash of the 404 page when creating a new document here https://nextjs.org/docs/basic-features/data-fetching#fallback-blocking